### PR TITLE
Cast SLEEP_TIME instead of parse it

### PR DESCRIPTION
**Pull request by M-Davies**
* Parsing appears to not exist as a method

Signed-off-by: Morgan Davies <morgandavies2020@gmail.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Review by karianna — APPROVED** [review]
LGTM

**Comment by karianna** [comment]
```
RepoHandlerTest > userDefaultsSetterAndGetterReturns() FAILED
    groovy.lang.MissingPropertyException: No such property: adoptJson for class: RepoHandlerTest
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:65)
        at org.codehaus.groovy.runtime.callsite.GetEffectivePogoPropertySite.getProperty(GetEffectivePogoPropertySite.java:87)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGroovyObjectGetProperty(AbstractCallSite.java:341)
        at RepoHandlerTest.userDefaultsSetterAndGetterReturns(RepoHandlerTest.groovy:125)

RepoHandlerTest > adoptDefaultsGetterReturns() FAILED
    groovy.lang.MissingMethodException: No signature of method: static org.junit.jupiter.api.Assertions.assertEquals() is applicable for argument types: (Boolean) values: [true]
    Possible solutions: assertEquals(byte, byte), assertEquals(byte, java.lang.Byte), assertEquals(char, char), assertEquals(char, java.lang.Character), assertEquals(double, double), assertEquals(double, java.lang.Double)
        at groovy.lang.MetaClassImpl.invokeStaticMissingMethod(MetaClassImpl.java:1568)
        at groovy.lang.MetaClassImpl.invokeStaticMethod(MetaClassImpl.java:1554)
        at org.codehaus.groovy.runtime.callsite.StaticMetaClassSite.call(StaticMetaClassSite.java:50)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
        at RepoHandlerTest.adoptDefaultsGetterReturns(RepoHandlerTest.groovy:67)
```

**Review by sxa — APPROVED** [review]
Looks plausible - let's give it a shot assuming the checks all pass :-)

File: pipelines/build/regeneration/build_job_generator.groovy [hunk 1]
```diff
@@ -166,7 +166,10 @@ node ("master") {
 
     def excludes = (params.EXCLUDES_LIST) ?: ""
     def jenkinsCreds = (params.JENKINS_AUTH) ?: ""
-    Integer sleepTime = (params.SLEEP_TIME) != "" ? Integer.parseInteger(SLEEP_TIME) : 900
+    Integer sleepTime = 900
+    if (params.SLEEP_TIME) {
+      sleepTime = SLEEP_TIME as Integer
+    }
```
Comment (AdamBrousseau) [inline pipelines/build/regeneration/build_job_generator.groovy:172]: Could you just change the original line to `? SLEEP_TIME.toInteger()` ?

Comment (M-Davies) [reply pipelines/build/regeneration/build_job_generator.groovy:172]: I remember trying that before when SLEEP_TIME was first implemented, it produced the same error

**Comment by M-Davies** [comment]
PR tester mac failures fixed by https://github.com/AdoptOpenJDK/openjdk-build/pull/2551 @sxa Feel free to merge when ready :)

File: pipelines/build/regeneration/build_job_generator.groovy (hunk 1 above)
Comment (AdamBrousseau) [reply pipelines/build/regeneration/build_job_generator.groovy:172]: Odd. We use it in the OpenJ9 scripts.
https://github.com/eclipse/openj9/search?q=toInteger%28%29&type=
Well whatever works. Tomato tomahtoe

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
Previously seen this error before on several other PR's. Seeing as this is breaking mac builds, I'll merge
```bash
00:17:56  checking for metal... [not found]
00:17:56  checking if metal can be run using xcrun... no
00:17:56  configure: error: XCode tool 'metal' neither found in path nor with xcrun
00:17:57  configure exiting with result code 1
00:17:57  Error: No configurations found for /Users/jenkins/workspace/build-scripts-pr-tester/build-test/jobs/jdk/jdk-mac-x64-hotspot/workspace/build/src.
00:17:57  Please run 'bash configure' to create a configuration.
```

**Pull request merged**